openquake/hazardlib/geo/surface/base.py

```diff
@@ -225,6 +239,79 @@ def get_joyner_boore_distance(self, mesh):
+            Admitted values for this parameter are 1 and 2. See description
+            below.
+        :returns:
+            Numpy array of distances in km.
+
+        See also :meth:`superclass method <.base.BaseSurface.get_ry0_distance>`
+        for spec of input and result values.
+
+        This method accepts an option id which determines the method to
+        be used for the calculation of the ry0 parameter. The default
+        method, the one selected with option 1, uses an average strike
+        direction. With option 2 the ry0 if computed using the lines
+        perpendicular to the two end segments of the set used to define
+        the fault trace.
+        """
+
+        if option == 1:
+
+            # This option computes ry0 by using an average strike direction
+            top_edge = self.get_mesh()[0:1]
+            mean_strike = self.get_strike()
+
+            dst1 = geodetic.distance_to_arc(top_edge.lons[0, 0],
+                                            top_edge.lats[0, 0],
+                                            (mean_strike+90.) % 360,
+                                            mesh.lons, mesh.lats)
+
+            dst2 = geodetic.distance_to_arc(top_edge.lons[0, -1],
+                                            top_edge.lats[0, -1],
+                                            (mean_strike+90.) % 360,
+                                            mesh.lons, mesh.lats)
+
+        elif option == 2:
+
+            # This option computes ry0 using the lines perpendicular to the
+            # segments at the two extremes of the fault trace
+            top_edge = self.get_mesh()[0:1]
+
+            # This computes the azimuth of the first segment of the trace
+            p_1 = Point(top_edge.lons[0, 0], top_edge.lats[0, 0],
+                        top_edge.depths[0, 0])
+            p_2 = Point(top_edge.lons[0, 1], top_edge.lats[0, 1],
+                        top_edge.depths[0, 1])
+            azimuth = p_1.azimuth(p_2)
+
+            dst1 = geodetic.distance_to_arc(p_1.longitude,
+                                            p_1.latitude,
+                                            (azimuth+90.) % 360,
+                                            mesh.lons, mesh.lats)
+
+            # This computes the azimuth of the last segment of the trace
+            p_1 = Point(top_edge.lons[0, -1], top_edge.lats[0, -1],
+                        top_edge.depths[0, -1])
+            p_2 = Point(top_edge.lons[0, -2], top_edge.lats[0, -2],
+                        top_edge.depths[0, -2])
+            azimuth = p_1.azimuth(p_2)
+
+            dst2 = geodetic.distance_to_arc(p_1.longitude,
+                                            p_1.latitude,
+                                            (azimuth+90.) % 360,
+                                            mesh.lons, mesh.lats)
+
+        # Get the shortest distance from two two lines
+        dst = numpy.fmin(numpy.abs(dst1), numpy.abs(dst2))
+
+        return dst
+
```

Comment:
Usually in such cases one define two methods, say get_ry0_distance_option_1 and get_ry0_distance_option_2 and call them inside get_ry0_distance, inside of having a big `if`.